text/0507-subnets.md

```diff
@@ -0,0 +1,402 @@
+vpc.addSecondaryAddressBlock(IpAddresses.ipv4('10.2.0.0/16'));
+```
+
+Or by providing an [IPAM] pool with a netmask length:
+
+```ts
+const ipam = new Ipam(stack, 'ipam');
+const pool = ipam.publicScope.addPool({
+  addressFamily: AddressFamily.IP_V4,
+  provisionedCidrs: [IpAddresses.ipv4('10.2.0.0/16')],
+});
+
+vpc.addSecondaryAddressBlock(IpAddresses.ipv4Ipam({
+  ipamPool: pool,
+  netmaskLength: 20
+}));
+```
+
+You can also add secondary IPv6 address blocks, in three different ways:
+
+```ts
+// 1. Providing an Ipv6 address block. Because IPv6 addresses are all publicly 
+// addressable, they must come from an address pool that you own and brought to 
+// AWS (BYOIP). So you must also provide the pool ID:
+vpc.addSecondaryAddressBlock(IpAddresses.ipv6({
+  cidr: '2001:db8:1234:1a00::/56',
+
+  // The pool of IPs you own. Not to be confused with an IPAM pool ID
+  poolId: 'my-ipv6-pool-id',
+}));
+
+// 2. Using an IPAM pool:
+vpc.addSecondaryAddressBlock(IpAddresses.ipv6Ipam({
+  ipamPoolId: pool,
+  netmaskLength: 64
+}));
+
+// 3. Using an Amazon-provided IPv6 CIDR block:
+vpc.addSecondaryAddressBlock(IpAddresses.amazonProvidedIpv6());
+```
+
+### Defining your own subnets
+
+By default, `Vpc` will create subnets for you (by default, a public and a
+private subnet per availability zone). If you want to define your own
+subnets, use the `addSubnet()` method, which will turn off this default
+behavior:
+
+```ts
+const subnet = vpc.addSubnet('subnet', {
+  cidrBlock: '10.2.0.0/20',
+  availabilityZone: 'us-west-2a'
+});
+```
+
+If you add more than one subnet to the VPC, the framework validates that there
+is no intersection between their address blocks. In addition, if all VPC IP
+address blocks (both primary and secondary) are provided as CIDR strings, the
+framework validates that each address block of all subnets is within one of the
+address blocks of the VPC.
+
+If you have added a secondary IPv6 block to your VPC, you can then add
+subnets with IPv6 ranges as well:
+
+```ts
+const subnet = vpc.addSubnet('subnet', {
+  cidrBlock: '2001:db8:1234:1a00::/60',
+  availabilityZone: 'us-west-2a'
+});
+```
+
+### Routing
+
+By default, `addSubnet()` creates isolated subnets, that only route traffic
+to other hosts inside the VPC. To define different routing policies for a
+subnet, provide a route table when creating it. For example, to create a
+public subnet:
+
+```ts
+const publicRouteTable = vpc.addRouteTable('routeTable', {
+  routes: [
+    // With no parameters, the destination is the entire VPC CIDR
+    Route.local(),
+
+    // By adding this route, all subnets that use this table become public
+    Route.to({
+      destination: '0.0.0.0/0',
+      target: Routers.INTERNET_GATEWAY,
+    }),
+  ],
+});
+
+const subnet = vpc.addSubnet('publicSubnet', {
+  cidrBlock: '10.2.0.0/20',
+  availabilityZone: 'us-west-2a',
+  routeTable: publicRouteTable,
+  mapPublicIpOnLaunch: false, // default: true for public subnets
+});
```

Comment:
Subnet can create its own route table by default, and then we can also do:

```ts
subnet.routeTable.addRoute(...)
```

Right? Let's support both and leave this pattern for people who want to have exactly one route table the same for all subnets?

Reply:
Agreed.